### PR TITLE
Feature/cc 7292 insights logging

### PR DESCRIPTION
**Pull request by dpetzold**

### CODE REVIEW

**Comment by dpetzold** [comment]
@gposton @grahamgreen @rbachman can you guys check this? All the work is in one commit https://github.com/clearcare/django-cacheops/commit/d483c7f9abe834ba60a22c3e855874831fbc5eef. It will send the following data to Insights:

**Comment by dpetzold** [comment]
There is a thumb here

https://github.com/clearcare/django-cacheops/commit/8cf47f1dd8468d8b7d2ff7941a7a9a41e94733dd#commitcomment-12905084

File: cacheops/__init__.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 VERSION = (1, 3, 1)
 __version__ = '.'.join(map(str, VERSION if VERSION[-1] else VERSION[:2]))
 
-from .simple import *
-from .query import *
-from .invalidation import *
+from .simple import *  # NOQA
+from .query import *  # NOQA
+from .invalidation import *  # NOQA
```
Comment (brianz) [inline cacheops/__init__.py]: Can you nuke the `# NOQA` thing?

**Comment by brianz** [comment]
Looks good.  It'd be cool to refactor this so that we could add any type of arbitrary reporting backend, be it NR or something else.

**Comment by dpetzold** [comment]
Yeah for sure. I was going to create a base class but just wanted to keep it simple for now.

**Comment by gposton** [comment]
Looks good!. One thing to be aware of is that insights metrics will 'roll out' as we hit the 300M event threshold.... and this is obviously going to send A LOT of events.  We'll want to turn it off (which we can do through NRs UI here: https://insights.newrelic.com/accounts/92732/manage/summary) and only enable it as needed.

**Comment by gposton** [comment]
Also, smart move making the app-name is based on the Django model... that will allow us to be very specific about what we enable/disable as the enable/disable functionality is tied to the app-name.

**Comment by gposton** [comment]
@dpetzold, not sure how the cachops-upgrade is planned, are we just merging their head into our fork?  I just want to ensure we won't loose this logging when we upgrade?

**Comment by dpetzold** [comment]
Thanks Glenn! Can't wait to get this and more metric into insights. This change will have to backported to the new cacheops which has a fairly different codebase. Though its only a 3 loc change in cacheops proper. I can take a look at that this week.

**Comment by gposton** [comment]
I wonder if it might be more likely to be accepted upstream if we design some way to hook into cachops events in a way that will allow users of cachops to execute local code on cachops events.  I'm thinking something like building a callback hook into cachops.  Then we can execute custom code whenever a cachops event (hit, miss, etc.) occurs.  That said, I'm not familiar w/ their code base and that may be asking a lot.... what do you think?

**Comment by dpetzold** [comment]
Yeah that sounds great. The code cacheops is calling should live the clearcare code base. Will make a pr for cacheops master :).

**Pull request closed without merging**